src/main/java/ch/njol/skript/hooks/Hook.java

```diff
@@ -18,70 +18,13 @@
  */
 package ch.njol.skript.hooks;
 
-import java.io.IOException;
+import org.jetbrains.annotations.ApiStatus;
 
-import ch.njol.skript.doc.Documentation;
-import org.bukkit.Bukkit;
-import org.bukkit.plugin.Plugin;
+@ApiStatus.Internal
+public interface Hook {
 
-import ch.njol.skript.Skript;
-import ch.njol.skript.localization.ArgsMessage;
+	boolean init();
 
-/**
- * @author Peter Güttinger
- */
-public abstract class Hook<P extends Plugin> {
-	
-	private final static ArgsMessage m_hooked = new ArgsMessage("hooks.hooked"),
-			m_hook_error = new ArgsMessage("hooks.error");
-	
-	public final P plugin;
-	
-	public final P getPlugin() {
-		return plugin;
-	}
-	
-	@SuppressWarnings("null")
-	public Hook() throws IOException {
-		@SuppressWarnings("unchecked")
-		final P p = (P) Bukkit.getPluginManager().getPlugin(getName());
-		plugin = p;
-		if (p == null) {
-			if (Documentation.canGenerateUnsafeDocs()) {
-				loadClasses();
-				if (Skript.logHigh())
-					Skript.info(m_hooked.toString(getName()));
-			}
-			return;
-		}
-
-		if (!init()) {
-			Skript.error(m_hook_error.toString(p.getName()));
-			return;
-		}
-
-		loadClasses();
-
-		if (Skript.logHigh())
-			Skript.info(m_hooked.toString(p.getName()));
+	String name();
```

Comment:
Skript has typically followed a get/set prefix style. Any particular reason for leaving it out here?